src/menu/ide_menu.rs

```diff
@@ -0,0 +1,1108 @@
+                " ".repeat(padding_right),
+                vertical_border,
+            )
+        }
+    }
+}
+
+impl Menu for IdeMenu {
+    /// Menu name
+    fn name(&self) -> &str {
+        self.name.as_str()
+    }
+
+    /// Menu indicator
+    fn indicator(&self) -> &str {
+        self.marker.as_str()
+    }
+
+    /// Deactivates context menu
+    fn is_active(&self) -> bool {
+        self.active
+    }
+
+    /// The ide menu can to quick complete if there is only one element
+    fn can_quick_complete(&self) -> bool {
+        true
+    }
+
+    fn can_partially_complete(
+        &mut self,
+        values_updated: bool,
+        editor: &mut Editor,
+        completer: &mut dyn Completer,
+    ) -> bool {
+        // If the values were already updated (e.g. quick completions are true)
+        // there is no need to update the values from the menu
+        if !values_updated {
+            self.update_values(editor, completer);
+        }
+
+        let values = self.get_values();
+        if let (Some(Suggestion { value, span, .. }), Some(index)) = find_common_string(values) {
+            let index = index.min(value.len());
+            let matching = &value[0..index];
+
+            // make sure that the partial completion does not overwrite user entered input
+            let extends_input = matching.starts_with(&editor.get_buffer()[span.start..span.end]);
+
+            if !matching.is_empty() && extends_input {
+                let mut line_buffer = editor.line_buffer().clone();
+                line_buffer.replace_range(span.start..span.end, matching);
+
+                let offset = if matching.len() < (span.end - span.start) {
+                    line_buffer
+                        .insertion_point()
+                        .saturating_sub((span.end - span.start) - matching.len())
+                } else {
+                    line_buffer.insertion_point() + matching.len() - (span.end - span.start)
+                };
+
+                line_buffer.set_insertion_point(offset);
+                editor.set_line_buffer(line_buffer, UndoBehavior::CreateUndoPoint);
+
+                // The values need to be updated because the spans need to be
+                // recalculated for accurate replacement in the string
+                self.update_values(editor, completer);
+
+                true
+            } else {
+                false
+            }
+        } else {
+            false
+        }
+    }
+
+    /// Selects what type of event happened with the menu
+    fn menu_event(&mut self, event: MenuEvent) {
+        match &event {
+            MenuEvent::Activate(_) => self.active = true,
+            MenuEvent::Deactivate => {
+                self.active = false;
+                self.input = None;
+            }
+            _ => {}
+        }
+
+        self.event = Some(event);
+    }
+
+    /// Update menu values
+    fn update_values(&mut self, editor: &mut Editor, completer: &mut dyn Completer) {
```

Comment:
By the way, the columnar menu was [recently updated](https://github.com/nushell/reedline/commit/0c5f9817465df3c80e062dd1b5b05bdd91924b83#diff-3fc955a6de109c1bf50c491b78e1794fcff831e06bba9d4f2455c8d68e5fba1d) so that `only_buffer_difference: true` would work, so you should probably copy the `update_values` method over again. Can't wait to use this!